internal/consensus/replay.go

```diff
@@ -259,6 +259,13 @@ func (h *Handshaker) Handshake(ctx context.Context, appClient abciclient.Client)
 		"protocol-version", res.AppVersion,
 	)
 
+	// Check app version is matched with the genesis file.
```

Comment:
Hmm. Is this really right, though? Won't the version of the app change when the node is upgraded, and at that point be different from the genesis version?

Reply:
Ah, I thought when the network is doing the protocol upgrade. it will update the genesis? Or the tendermint always allows the app changes regarding the genesis changes or not?

The app_version explanation:
https://github.com/tendermint/tendermint/blob/master/spec/core/genesis.md

Reply:
okay, the app soft upgrade can update the app version without changing the genesis. That made me confused why the app_version had been put into the genesis but not using it to validate the app version.

Reply:
close this PR and create an issue https://github.com/tendermint/tendermint/issues/8095